Source/Particles/PhysicalParticleContainer.cpp

```diff
@@ -309,6 +309,23 @@ PhysicalParticleContainer::AddGaussianBeam(Real x_m, Real y_m, Real z_m,
                   1, particle_w.dataPtr(),1);
 }
 
+void
+PhysicalParticleContainer::AddExternalFileBeam(std::string s_f)
```

Comment:
```suggestion
PhysicalParticleContainer::AddExternalFileBeam(const std::string s_f)
```

Reply:
Thanks for checking this @ax3l ! Just because I am still having a hard time understanding when we choose to use const. You just want to make sure we don't change the file name inside the function, right?

Reply:
`const` can mean a number of different things, as shown for instance in https://www.studytonight.com/cpp/const-keyword.php (but @ax3l probably has better refs). While the code will usually work without adding `const`, we do it for two main reasons:
- let the compiler know that something should not be modified (written into), so it can make further optimizations
- let other developers know our intent (and let the compiler give an error if we accidentally try to modify it).

In general, giving more information to the compiler (e.g., with keywords and qualifiers `const` `constexpr` `noexcept` `override` etc.) is always a good thing for both performance and code safety.